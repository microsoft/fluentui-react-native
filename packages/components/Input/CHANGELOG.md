# Change Log - @fluentui-react-native/input

This log was last generated on Thu, 09 Nov 2023 02:39:09 GMT and should not be manually modified.

<!-- Start content -->

## 0.5.0

Thu, 09 Nov 2023 02:39:09 GMT

### Minor changes

- Update to react-native 0.72 (30809111+acoates-ms@users.noreply.github.com)
- Bump @fluentui-react-native/framework to v0.12.0
- Bump @fluentui-react-native/icon to v0.20.0
- Bump @fluentui-react-native/interactive-hooks to v0.25.0
- Bump @fluentui-react-native/styling-utils to v0.6.0
- Bump @fluentui-react-native/text to v0.22.0
- Bump @fluentui-react-native/theme-tokens to v0.26.0
- Bump @fluentui-react-native/tokens to v0.22.0
- Bump @fluentui-react-native/use-styling to v0.12.0

## 0.4.9

Tue, 07 Nov 2023 20:20:34 GMT

### Patches

- Manually bump packages stuck in the pipeline (safreibe@microsoft.com)
- Bump @fluentui-react-native/framework to v0.11.10
- Bump @fluentui-react-native/icon to v0.19.16
- Bump @fluentui-react-native/interactive-hooks to v0.24.12
- Bump @fluentui-react-native/text to v0.21.14
- Bump @fluentui-react-native/theme-tokens to v0.25.4
- Bump @fluentui-react-native/tokens to v0.21.6

## 0.4.7

Mon, 30 Oct 2023 21:26:06 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.11.8
- Bump @fluentui-react-native/icon to v0.19.14
- Bump @fluentui-react-native/interactive-hooks to v0.24.10
- Bump @fluentui-react-native/text to v0.21.12
- Bump @fluentui-react-native/theme-tokens to v0.25.2
- Bump @fluentui-react-native/tokens to v0.21.4

## 0.4.6

Wed, 25 Oct 2023 20:50:34 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.11.7
- Bump @fluentui-react-native/icon to v0.19.13
- Bump @fluentui-react-native/interactive-hooks to v0.24.9
- Bump @fluentui-react-native/text to v0.21.11

## 0.4.5

Wed, 25 Oct 2023 17:21:07 GMT

### Patches

- Bump @fluentui-react-native/icon to v0.19.12

## 0.4.4

Mon, 25 Sep 2023 12:49:34 GMT

### Patches

- Bump @fluentui-react-native/icon to v0.19.11
- Bump @fluentui-react-native/text to v0.21.10

## 0.4.3

Tue, 29 Aug 2023 22:56:32 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.11.6
- Bump @fluentui-react-native/icon to v0.19.10
- Bump @fluentui-react-native/interactive-hooks to v0.24.8
- Bump @fluentui-react-native/text to v0.21.9
- Bump @fluentui-react-native/tokens to v0.21.3

## 0.4.2

Tue, 29 Aug 2023 21:59:04 GMT

### Patches

- Bump @fluentui-react-native/icon to v0.19.9

## 0.4.1

Wed, 23 Aug 2023 18:55:24 GMT

### Patches

- Update SVG dependencies (ruaraki@microsoft.com)
- Bump @fluentui-react-native/icon to v0.19.8
- Bump @fluentui-react-native/text to v0.21.8

## 0.4.0

Mon, 21 Aug 2023 22:45:19 GMT

### Minor changes

- Tokenize for iOS and add to tester (ayushsinghs@yahoo.in)

## 0.3.9

Mon, 14 Aug 2023 20:14:41 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.11.5
- Bump @fluentui-react-native/icon to v0.19.7
- Bump @fluentui-react-native/interactive-hooks to v0.24.7
- Bump @fluentui-react-native/text to v0.21.7
- Bump @fluentui-react-native/tokens to v0.21.2

## 0.3.8

Wed, 09 Aug 2023 22:02:15 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.11.4
- Bump @fluentui-react-native/icon to v0.19.6
- Bump @fluentui-react-native/interactive-hooks to v0.24.6
- Bump @fluentui-react-native/text to v0.21.6
- Bump @fluentui-react-native/theme-tokens to v0.25.1
- Bump @fluentui-react-native/use-styling to v0.11.1

## 0.3.7

Mon, 31 Jul 2023 19:29:59 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.11.3
- Bump @fluentui-react-native/icon to v0.19.5
- Bump @fluentui-react-native/interactive-hooks to v0.24.5
- Bump @fluentui-react-native/text to v0.21.5

## 0.3.6

Tue, 18 Jul 2023 19:09:32 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.11.2
- Bump @fluentui-react-native/icon to v0.19.4
- Bump @fluentui-react-native/interactive-hooks to v0.24.4
- Bump @fluentui-react-native/text to v0.21.4

## 0.3.5

Fri, 07 Jul 2023 20:45:12 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.11.1
- Bump @fluentui-react-native/icon to v0.19.3
- Bump @fluentui-react-native/interactive-hooks to v0.24.3
- Bump @fluentui-react-native/text to v0.21.3
- Bump @fluentui-react-native/tokens to v0.21.1

## 0.3.4

Mon, 19 Jun 2023 18:37:35 GMT

### Patches

- Bump @fluentui-react-native/icon to v0.19.2
- Bump @fluentui-react-native/interactive-hooks to v0.24.2
- Bump @fluentui-react-native/text to v0.21.2

## 0.3.3

Sat, 17 Jun 2023 00:00:18 GMT

### Patches

- Bump @fluentui-react-native/icon to v0.19.1
- Bump @fluentui-react-native/interactive-hooks to v0.24.1
- Bump @fluentui-react-native/text to v0.21.1

## 0.3.2

Mon, 12 Jun 2023 17:46:44 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.11.0
- Bump @fluentui-react-native/icon to v0.19.0
- Bump @fluentui-react-native/interactive-hooks to v0.24.0
- Bump @fluentui-react-native/text to v0.21.0
- Bump @fluentui-react-native/use-styling to v0.11.0

## 0.3.1

Fri, 09 Jun 2023 15:29:07 GMT

### Patches

- Bump @fluentui-react-native/icon to v0.18.1
- Bump @fluentui-react-native/text to v0.20.1

## 0.3.0

Mon, 05 Jun 2023 19:26:23 GMT

### Minor changes

- Upgrade to React Native 0.71 (sanajmi@microsoft.com)
- Bump @fluentui-react-native/framework to v0.10.0
- Bump @fluentui-react-native/icon to v0.18.0
- Bump @fluentui-react-native/interactive-hooks to v0.23.0
- Bump @fluentui-react-native/styling-utils to v0.5.0
- Bump @fluentui-react-native/text to v0.20.0
- Bump @fluentui-react-native/theme-tokens to v0.25.0
- Bump @fluentui-react-native/tokens to v0.21.0
- Bump @fluentui-react-native/use-styling to v0.10.0

## 0.2.3

Tue, 30 May 2023 20:34:15 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.9.10
- Bump @fluentui-react-native/icon to v0.17.27
- Bump @fluentui-react-native/interactive-hooks to v0.22.31
- Bump @fluentui-react-native/text to v0.19.33
- Bump @fluentui-react-native/theme-tokens to v0.24.7
- Bump @fluentui-react-native/tokens to v0.20.16

## 0.2.2

Wed, 10 May 2023 11:59:16 GMT

### Patches

- Change svg to component (ayushsinghs@yahoo.in)

## 0.2.1

Wed, 03 May 2023 07:08:52 GMT

### Patches

- Bump @fluentui-react-native/framework to v0.9.9
- Bump @fluentui-react-native/icon to v0.17.26
- Bump @fluentui-react-native/interactive-hooks to v0.22.30
- Bump @fluentui-react-native/text to v0.19.32
- Bump @fluentui-react-native/theme-tokens to v0.24.6
- Bump @fluentui-react-native/tokens to v0.20.15

## 0.2.0

Thu, 20 Apr 2023 09:41:22 GMT

### Minor changes

- Expand input functionality (ayushsinghs@yahoo.in)

## 0.1.2

Wed, 12 Apr 2023 18:42:22 GMT

### Patches

- Fix the pipeline (ruaraki@microsoft.com)

## 0.1.1

Tue, 11 Apr 2023 10:42:50 GMT

### Patches

- Added new component - input (ayushsinghs@yahoo.in)
- Bump @fluentui-react-native/framework to v0.9.8
- Bump @fluentui-react-native/icon to v0.17.25
- Bump @fluentui-react-native/interactive-hooks to v0.22.29
- Bump @fluentui-react-native/text to v0.19.31
- Bump @fluentui-react-native/tokens to v0.20.14
- Bump @fluentui-react-native/use-styling to v0.9.4
