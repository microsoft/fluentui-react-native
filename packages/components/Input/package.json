{
  "name": "@fluentui-react-native/input",
  "version": "0.3.4",
  "description": "A cross-platform Text input component using the Fluent Design System",
  "main": "src/index.ts",
  "module": "src/index.ts",
  "typings": "lib/index.d.ts",
  "onPublish": {
    "main": "lib-commonjs/index.js",
    "module": "lib/index.js"
  },
  "scripts": {
    "build": "fluentui-scripts build",
    "clean": "fluentui-scripts clean",
    "depcheck": "fluentui-scripts depcheck",
    "just": "fluentui-scripts",
    "lint": "fluentui-scripts eslint",
    "test": "fluentui-scripts jest",
    "update-snapshots": "fluentui-scripts jest -u",
    "prettier": "fluentui-scripts prettier",
    "prettier-fix": "fluentui-scripts prettier --fix true"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui-react-native.git",
    "directory": "packages/components/Input"
  },
  "dependencies": {
    "@fluentui-react-native/framework": "0.11.0",
    "@fluentui-react-native/icon": "^0.19.2",
    "@fluentui-react-native/interactive-hooks": ">=0.24.2 <1.0.0",
    "@fluentui-react-native/styling-utils": ">=0.5.0 <1.0.0",
    "@fluentui-react-native/text": ">=0.21.2 <1.0.0",
    "@fluentui-react-native/theme-tokens": "^0.25.0",
    "@fluentui-react-native/tokens": ">=0.21.0 <1.0.0",
    "@fluentui-react-native/use-styling": ">=0.11.0 <1.0.0"
  },
  "devDependencies": {
    "@fluentui-react-native/eslint-config-rules": "^0.1.1",
    "@fluentui-react-native/scripts": "^0.1.1",
    "@fluentui-react-native/test-tools": ">=0.1.1 <1.0.0",
    "react": "18.2.0",
    "react-native": "^0.71.0",
    "react-native-svg": "^13.7.0"
  },
  "peerDependencies": {
    "react": "18.2.0",
    "react-native": "^0.71.0",
    "react-native-svg": "^13.7.0"
  },
  "author": "",
  "license": "MIT",
  "rnx-kit": {
    "kitType": "library",
    "alignDeps": {
      "presets": [
        "microsoft/react-native"
      ],
      "requirements": [
        "react-native@0.71"
      ],
      "capabilities": [
        "core",
        "core-android",
        "core-ios",
        "react",
        "svg"
      ]
    }
  }
}
