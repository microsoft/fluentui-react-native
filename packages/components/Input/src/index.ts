export { Input, inputLookup } from './Input';
export type { InputProps, InputSlotProps, InputTokens, InputType } from './Input.types';
export { input } from './Input.types';
export { DismissSvg } from './assets/dismissSvg';
