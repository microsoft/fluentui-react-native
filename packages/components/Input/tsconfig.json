{
  "extends": "@fluentui-react-native/scripts/configs/tsconfig.json",
  "compilerOptions": {
    "outDir": "lib"
  },
  "include": ["src", "svgs.d.ts"]
}
