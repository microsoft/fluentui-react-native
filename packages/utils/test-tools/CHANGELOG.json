{
  "name": "@fluentui-react-native/test-tools",
  "entries": [
    {
      "date": "Thu, 26 Aug 2021 04:48:40 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.3",
            "author": "67026167+chiuam@users.noreply.github.com",
            "commit": "b8c866644b2c5b64530990e59153424b0d6f5dcc",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Wed, 18 Aug 2021 23:00:57 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.2",
            "author": "ruaraki@microsoft.com",
            "commit": "345d1de930ff42359260da7f1e68a16059f4d763",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Fri, 13 Aug 2021 16:12:26 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.1",
            "author": "t-lindaweng@microsoft.com",
            "commit": "8946966df6ddc0ef1c0b43e576a4124c678d7c89",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Thu, 12 Aug 2021 17:23:44 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.11.0",
            "author": "ruaraki@microsoft.com",
            "commit": "e524662e3e40fe45a2dcc6e2acd3aca1197b5b5a",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Tue, 10 Aug 2021 21:12:30 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.10.0",
            "author": "ruaraki@microsoft.com",
            "commit": "ce49f33c08add7892191cf840d2fdcccfd0a16bb",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Sat, 07 Aug 2021 00:40:04 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "none": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.3",
            "author": "jasonmo@microsoft.com",
            "commit": "e506337e109ef24e675c2ac794179db86fcb7e86",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Fri, 06 Aug 2021 00:36:02 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "none": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.3",
            "author": "ruaraki@microsoft.com",
            "commit": "56ccce452d0dbbd5c24f53ebdd467f2d01dda694",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Wed, 04 Aug 2021 06:26:25 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.3",
            "author": "67026167+chiuam@users.noreply.github.com",
            "commit": "68dd61bd45ca8ffacc1580e468f3c01810e3093e",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Tue, 27 Jul 2021 22:17:20 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.2",
            "author": "jasonmo@microsoft.com",
            "commit": "1cd8611e1e88848ab66f7c0e9113c946f968260c",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Tue, 27 Jul 2021 18:55:58 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "none": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.1",
            "author": "4123478+tido64@users.noreply.github.com",
            "commit": "07d04d166b13ff79e07a8c4cbc2e06afb1e4c3e6",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    },
    {
      "date": "Mon, 26 Jul 2021 20:41:04 GMT",
      "tag": "@fluentui-react-native/test-tools_v0.1.1",
      "version": "0.1.1",
      "comments": {
        "patch": [
          {
            "comment": "Bump @fluentui-react-native/theme-types to v0.9.1",
            "author": "jasonmo@microsoft.com",
            "commit": "ad02fbbe9eaa5fbe6ec6126ccb719a1280f108a6",
            "package": "@fluentui-react-native/test-tools"
          }
        ]
      }
    }
  ]
}
