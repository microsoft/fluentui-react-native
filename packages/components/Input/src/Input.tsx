/** @jsxRuntime classic */
/** @jsx withSlots */
import { Pressable, ScrollView, TextInput, View } from 'react-native';

import type { UseSlots } from '@fluentui-react-native/framework';
import { compose } from '@fluentui-react-native/framework';
import { Icon } from '@fluentui-react-native/icon';
import { Text } from '@fluentui-react-native/text';

import { stylingSettings } from './Input.styling';
import type { InputType, InputProps } from './Input.types';
import { input } from './Input.types';

/**
 * A function which determines if a set of styles should be applied to the component given the current state and props of the input.
 *
 * @param layer The name of the state that is being checked for
 * @param userProps The props that were passed into the input
 * @returns Whether the styles that are assigned to the layer should be applied to the input
 */
export const inputLookup = () => null;

export const Input = compose<InputType>({
  displayName: input,
  ...stylingSettings,
  slots: {
    root: ScrollView,
    icon: Icon,
    inputWrapper: View,
    label: Text,
    input: View,
    textInput: TextInput,
    accessoryIconPressable: Pressable,
    accessoryIcon: Icon,
    assistiveText: Text,
    accessoryText: Text,
  },
  useRender: (_userProps: InputProps, _useSlots: UseSlots<InputType>) => {
    return (_final: InputProps) => {
      return null;
    };
  },
});
