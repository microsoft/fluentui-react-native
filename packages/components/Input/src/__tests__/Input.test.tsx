import * as React from 'react';

import { checkRenderConsistency, checkReRender } from '@fluentui-react-native/test-tools';
import * as renderer from 'react-test-renderer';

import { Input } from '../Input';

const fontBuiltInProps = {
  fontFamily: 'Arial',
  codepoint: 0x2663,
  fontSize: 16,
};
const iconProps = { fontSource: { ...fontBuiltInProps }, color: '#fff' };

describe('Input component tests', () => {
  it('Input default', () => {
    const tree = renderer.create(<Input accessoryIcon={iconProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Input with placeholder', () => {
    const tree = renderer.create(<Input placeholder="Test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Input without accessoryIcon', () => {
    const tree = renderer.create(<Input placeholder="Test" accessoryIcon={null} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Input with all optional text', () => {
    const tree = renderer.create(<Input placeholder="Test" accessoryText="Accessory" label="Label" assistiveText="Assistive" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Input with icon', () => {
    const tree = renderer.create(<Input placeholder="Test" defaultIcon={iconProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Input in error state', () => {
    const tree = renderer.create(<Input error="Error" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Input simple rendering does not invalidate styling', () => {
    checkRenderConsistency(() => <Input />, 2);
  });

  it('Input re-renders correctly', () => {
    checkReRender(() => <Input />, 2);
  });

  it('Input with placeholder', () => {
    checkReRender(() => <Input placeholder="Test" accessoryIcon={null} />, 2);
  });
});
