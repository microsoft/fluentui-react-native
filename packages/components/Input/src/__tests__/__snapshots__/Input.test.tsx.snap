// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component tests Input default 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "flex-start",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
  defaultIconProps={null}
  focusedIconProps={null}
  keyboardShouldPersistTaps="handled"
  onBlur={[Function]}
  onFocus={[Function]}
  onHoverIn={[Function]}
  onHoverOut={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  scrollEnabled={false}
  setIconProps={[Function]}
  setText={[Function]}
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderBottomColor": "#d1d1d1",
          "borderBottomWidth": 0.5,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 12,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholderTextColor="#616161"
        style={
          {
            "color": "#616161",
            "dynamicTypeRamp": undefined,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Segoe UI",
            "fontSize": 13,
            "fontStyle": undefined,
            "fontWeight": "400",
            "letterSpacing": undefined,
            "lineHeight": undefined,
            "margin": 0,
            "marginBottom": 0,
            "marginEnd": 0,
            "marginStart": 0,
            "marginTop": 0,
            "maximumFontSize": undefined,
            "padding": 0,
            "textDecorationLine": undefined,
          }
        }
        value=""
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`Input component tests Input in error state 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "flex-start",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
  defaultIconProps={null}
  focusedIconProps={null}
  keyboardShouldPersistTaps="handled"
  onBlur={[Function]}
  onFocus={[Function]}
  onHoverIn={[Function]}
  onHoverOut={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  scrollEnabled={false}
  setIconProps={[Function]}
  setText={[Function]}
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderBottomColor": "#bc2f32",
          "borderBottomWidth": 0.5,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 12,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholderTextColor="#242424"
        selectionColor="#808080"
        style={
          {
            "color": "#242424",
            "dynamicTypeRamp": undefined,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Segoe UI",
            "fontSize": 13,
            "fontStyle": undefined,
            "fontWeight": "400",
            "letterSpacing": undefined,
            "lineHeight": undefined,
            "margin": 0,
            "marginBottom": 0,
            "marginEnd": 0,
            "marginStart": 0,
            "marginTop": 0,
            "maximumFontSize": undefined,
            "padding": 0,
            "textDecorationLine": undefined,
          }
        }
        value=""
      />
    </View>
    <Text
      style={
        {
          "color": "#bc2f32",
          "margin": 0,
          "marginBottom": 0,
          "marginEnd": 0,
          "marginStart": 0,
          "marginTop": 0,
          "paddingVertical": 4,
        }
      }
    >
      Error
    </Text>
  </View>
</RCTScrollView>
`;

exports[`Input component tests Input with all optional text 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "flex-start",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
  defaultIconProps={null}
  focusedIconProps={null}
  keyboardShouldPersistTaps="handled"
  onBlur={[Function]}
  onFocus={[Function]}
  onHoverIn={[Function]}
  onHoverOut={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  scrollEnabled={false}
  setIconProps={[Function]}
  setText={[Function]}
>
  <View>
    <Text
      style={
        {
          "color": "#616161",
          "margin": 0,
          "marginBottom": 0,
          "marginEnd": 0,
          "marginStart": 0,
          "marginTop": 12,
        }
      }
    >
      Label
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderBottomColor": "#d1d1d1",
          "borderBottomWidth": 0.5,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 12,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Test"
        placeholderTextColor="#616161"
        style={
          {
            "color": "#616161",
            "dynamicTypeRamp": undefined,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Segoe UI",
            "fontSize": 13,
            "fontStyle": undefined,
            "fontWeight": "400",
            "letterSpacing": undefined,
            "lineHeight": undefined,
            "margin": 0,
            "marginBottom": 0,
            "marginEnd": 0,
            "marginStart": 0,
            "marginTop": 0,
            "maximumFontSize": undefined,
            "padding": 0,
            "textDecorationLine": undefined,
          }
        }
        value=""
      />
    </View>
    <Text
      style={
        {
          "color": "#616161",
          "margin": 0,
          "marginBottom": 0,
          "marginEnd": 0,
          "marginStart": 0,
          "marginTop": 0,
          "paddingVertical": 4,
        }
      }
    >
      Assistive
    </Text>
  </View>
</RCTScrollView>
`;

exports[`Input component tests Input with icon 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "flex-start",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
  defaultIconProps={
    {
      "color": "#fff",
      "fontSource": {
        "codepoint": 9827,
        "fontFamily": "Arial",
        "fontSize": 16,
      },
    }
  }
  focusedIconProps={null}
  keyboardShouldPersistTaps="handled"
  onBlur={[Function]}
  onFocus={[Function]}
  onHoverIn={[Function]}
  onHoverOut={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  scrollEnabled={false}
  setIconProps={[Function]}
  setText={[Function]}
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={false}
        style={
          {
            "color": "#fff",
            "fontFamily": "Arial",
            "fontSize": 16,
            "fontWeight": "400",
            "margin": 0,
          }
        }
      >
        â™£
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "borderBottomColor": "#d1d1d1",
            "borderBottomWidth": 0.5,
            "display": "flex",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingVertical": 12,
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Test"
          placeholderTextColor="#616161"
          style={
            {
              "color": "#616161",
              "dynamicTypeRamp": undefined,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "Segoe UI",
              "fontSize": 13,
              "fontStyle": undefined,
              "fontWeight": "400",
              "letterSpacing": undefined,
              "lineHeight": undefined,
              "margin": 0,
              "marginBottom": 0,
              "marginEnd": 0,
              "marginStart": 0,
              "marginTop": 0,
              "maximumFontSize": undefined,
              "padding": 0,
              "textDecorationLine": undefined,
            }
          }
          value=""
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`Input component tests Input with placeholder 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "flex-start",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
  defaultIconProps={null}
  focusedIconProps={null}
  keyboardShouldPersistTaps="handled"
  onBlur={[Function]}
  onFocus={[Function]}
  onHoverIn={[Function]}
  onHoverOut={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  scrollEnabled={false}
  setIconProps={[Function]}
  setText={[Function]}
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderBottomColor": "#d1d1d1",
          "borderBottomWidth": 0.5,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 12,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Test"
        placeholderTextColor="#616161"
        style={
          {
            "color": "#616161",
            "dynamicTypeRamp": undefined,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Segoe UI",
            "fontSize": 13,
            "fontStyle": undefined,
            "fontWeight": "400",
            "letterSpacing": undefined,
            "lineHeight": undefined,
            "margin": 0,
            "marginBottom": 0,
            "marginEnd": 0,
            "marginStart": 0,
            "marginTop": 0,
            "maximumFontSize": undefined,
            "padding": 0,
            "textDecorationLine": undefined,
          }
        }
        value=""
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`Input component tests Input without accessoryIcon 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "alignItems": "flex-start",
      "alignSelf": "flex-start",
      "backgroundColor": "#ffffff",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
  defaultIconProps={null}
  focusedIconProps={null}
  keyboardShouldPersistTaps="handled"
  onBlur={[Function]}
  onFocus={[Function]}
  onHoverIn={[Function]}
  onHoverOut={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  scrollEnabled={false}
  setIconProps={[Function]}
  setText={[Function]}
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "borderBottomColor": "#d1d1d1",
          "borderBottomWidth": 0.5,
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 12,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Test"
        placeholderTextColor="#616161"
        style={
          {
            "color": "#616161",
            "dynamicTypeRamp": undefined,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontFamily": "Segoe UI",
            "fontSize": 13,
            "fontStyle": undefined,
            "fontWeight": "400",
            "letterSpacing": undefined,
            "lineHeight": undefined,
            "margin": 0,
            "marginBottom": 0,
            "marginEnd": 0,
            "marginStart": 0,
            "marginTop": 0,
            "maximumFontSize": undefined,
            "padding": 0,
            "textDecorationLine": undefined,
          }
        }
        value=""
      />
    </View>
  </View>
</RCTScrollView>
`;
